rpcserver.go

```diff
@@ -510,17 +510,53 @@ func (s *rpcServer) InitAccount(ctx context.Context,
 			"must be specified")
 	}
 
-	// Determine the desired transaction fee.
-	confTarget := req.GetConfTarget()
-	if confTarget < 1 {
-		return nil, fmt.Errorf("confirmation target must be " +
-			"greater than 0")
+	if req.GetFeeRateSatPerKw() > 0 && req.GetConfTarget() > 0 {
+		return nil, fmt.Errorf("you must set only one of the sat/vbyte " +
+			"and confirmation target parameters")
+	}
+
+	var feeRate chainfee.SatPerKWeight
+
+	switch {
+	case req.GetFeeRateSatPerKw() > 0:
+		feeRate = chainfee.SatPerKWeight(req.GetFeeRateSatPerKw())
+
+	case req.GetConfTarget() > 0:
+		// Determine the desired transaction fee.
+		value := btcutil.Amount(req.AccountValue)
+		confTarget := req.GetConfTarget()
+
+		feeRate, _, err := s.accountManager.QuoteAccount(
+			ctx, value, confTarget,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("unable to estimate on-chain fees: "+
+				"%v", err)
+		}
+
+		// If the fee estimation ever returns a value too small
+		// we set it to a valid minimum
+		if feeRate < chainfee.FeePerKwFloor {
+			feeRate = chainfee.FeePerKwFloor
+		}
+
+		log.Infof("Estimated total chain fee of %v for new account with "+
+			"value=%v, conf_target=%v", feeRate, value, confTarget)
+
+	default:
+		return nil, fmt.Errorf("either sat/vbyte or confirmation target " +
```

Comment:
sats/kw ?

Reply:
Maybe I'd move the error case from the above `if` statement here too as a case.

Reply:
Let me add another small PR with this fix + adding the same check for `expiryHeight`